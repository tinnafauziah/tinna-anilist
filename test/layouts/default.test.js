import defaultLayout from '~/layouts/default';

const initialMocks = {
  $router: {
    push: jest.fn(),
  },
  $route: {
    path: '/',
  },
};

describe('Layout Default', () => {
  const mockPage = (customMocks) => {
    const mocks = { ...initialMocks, ...customMocks }
    return global.shallowMount(defaultLayout, { mocks });
  };
  it.each([['/'], ['/authorize']])('should layout page match snapshot when page is %s', () => {
    const customPath = {
      $route: {
        path: '/',
      }
    }
    const wrapper = mockPage(customPath);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('should redirect to home', async () => {
    const wrapper = mockPage();
    const toolbarTitle = await wrapper.find('.toolbar-title');
    toolbarTitle.trigger('click');

    expect(initialMocks.$router.push).toBeCalledWith('/');
  });
});
