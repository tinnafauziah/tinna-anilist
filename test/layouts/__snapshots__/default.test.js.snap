// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Default should layout page match snapshot when page is / 1`] = `
<v-app-stub
  id="app"
>
  <v-main-stub
    tag="main"
  >
    <v-app-bar-stub
      color="indigo lighten-4"
      extensionheight="48"
      src=""
      tag="header"
      value="true"
    >
      <v-toolbar-title-stub
        class="toolbar-title"
        tag="div"
      >
        Anime List
      </v-toolbar-title-stub>
      <v-spacer-stub
        tag="div"
      />
      <v-btn-stub
        activeclass=""
        tag="button"
        type="button"
      >
        Login with Anilist
      </v-btn-stub>
    </v-app-bar-stub>
    <nuxt-stub
      class="mt-8 ml-8"
    />
  </v-main-stub>
</v-app-stub>
`;

exports[`Layout Default should layout page match snapshot when page is /authorize 1`] = `
<v-app-stub
  id="app"
>
  <v-main-stub
    tag="main"
  >
    <v-app-bar-stub
      color="indigo lighten-4"
      extensionheight="48"
      src=""
      tag="header"
      value="true"
    >
      <v-toolbar-title-stub
        class="toolbar-title"
        tag="div"
      >
        Anime List
      </v-toolbar-title-stub>
      <v-spacer-stub
        tag="div"
      />
      <v-btn-stub
        activeclass=""
        tag="button"
        type="button"
      >
        Login with Anilist
      </v-btn-stub>
    </v-app-bar-stub>
    <nuxt-stub
      class="mt-8 ml-8"
    />
  </v-main-stub>
</v-app-stub>
`;
